2048Solution/main/java/models/GameService.java

```diff
@@ -0,0 +1,230 @@
+package main.java.models;
+import java.util.ArrayList;
+import java.lang.Integer;
+public class GameService {
+    GameBoard board = new GameBoard();
+    GameUtility gameUtility = new GameUtility();
+    void initialize(int size){
+        board.setSize(size);
+        board.initializeBoard(size);
+        fillValueInBoard();
+        fillValueInBoard();
+    }
+
+    void fillValueInBoard(){
+        gameUtility.getRandomIndex(board);
+        board.setValueInBoard(board.getRowToBeFilled(),board.getColumnToBeFilled(), 2);
+    }
+    void printValues(){
+        for(int i = 0; i < board.getSize(); i++){
+            for(int j = 0; j < board.getSize(); j++){
+                if(board.getValueInBoard(i,j) == 0){
+                    System.out.print("- ");
+                }
+                else{
+                    System.out.print(board.getValueInBoard(i,j) + " ");
+                }
+            }
+            System.out.println();
+        }
+    }
+    void startGame(int direction){
+       makeDecision(direction);
+       fillValueInBoard();
+       printValues();
+    }
+    boolean hasWon(){
+        for(int i = 0; i< board.getSize(); i++){
+            for(int j=  0; j < board.getSize(); j++){
+                if(board.getValueInBoard(i,j) == 2048){
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    boolean isGameOver(){
+        for(int i = 0; i< board.getSize(); i++) {
+            for (int j = 0; j < board.getSize(); j++) {
+                if(board.getValueInBoard(i,j) == 0)
+                    return false;
+            }
+        }
+        return true;
+    }
+    void makeDecision(int direction){
+        switch (direction) {
+            case 0 -> moveAllColumnsLeft();
+            case 1 -> moveAllColumnRight();
+            case 2 -> moveAllRowsUp();
+            case 3 -> moveAllRowsDown();
+        }
+    }
+    private void moveAllRowsDown() {
```

Comment:
Make this function smaller by extracting functions

Reply:
Shall I extract a part of code, even after knowing that it cannot be reused, from this function?
If yes then for each of these "move.." function, if I extract 2 piece of code, i.e if I create 3 functions for every "move" operation , then  instead of 4 "move.." functions which we have currently our class would be having  12 functions.
Will it be fine if I do this?

Reply:
Yes. But see if those functions can be reused.